aws_lambda_powertools/utilities/idempotency/persistence/dynamodb.py

```diff
@@ -221,6 +222,15 @@ def _put_record(self, data_record: DataRecord) -> None:
             condition_expression = (
                 f"{idempotency_key_not_exist} OR {idempotency_expiry_expired} OR ({inprogress_expiry_expired})"
             )
+
+            # Use DynamoDB's ReturnValuesOnConditionCheckFailure to optimize put and get operations and optimize costs.
+            # This feature is supported in boto3 versions 1.26.164 and later.
+            support_return_on_check_failure = (
+                {"ReturnValuesOnConditionCheckFailure": "ALL_OLD"}
+                if self.boto3_supports_condition_check_failure(boto3.__version__)
+                else {}
+            )
```

Comment:
It seems to me this code is going to run on every single idemotent request. Can we memorize this check so we’re not wasting cycles all the time?

Reply:
Yes!! It makes perfect sense, I changed it to the Class init method, that way we keep it in memory.